src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -677,28 +677,17 @@ public static unsafe string Title
 
         public static void Beep()
         {
-            const char BellCharacter = '\u0007'; // Windows doesn't use terminfo, so the codepoint is hardcoded.
-
             if (!Console.IsOutputRedirected)
             {
-                Console.Out.Write(BellCharacter);
-                return;
-            }
-
-            if (!Console.IsErrorRedirected)
-            {
-                Console.Error.Write(BellCharacter);
-                return;
+                ReadOnlySpan<byte> bell = "\u0007"u8; // Windows doesn't use terminfo, so the codepoint is hardcoded.
+                int errorCode = WindowsConsoleStream.WriteFileNative(OutputHandle, bell, useFileAPIs: Console.OutputEncoding.CodePage != UnicodeCodePage);
```

Comment:
Every time `OutputHandle` is called, there would be a PInvoke to `GetStdHandle`. Should the value be cahced?

And I believe the bell char will be printed when the program is called thru `Program > somefile.txt` because the `>` syntax will alter the result of `GetStdHandle`. ([some quick search](https://stackoverflow.com/questions/9021916/how-do-i-check-if-my-delphi-console-app-is-redirected-to-a-file-or-pipe))

Reply:
> And I believe the bell char will be printed when the program is called thru Program > somefile.txt because the > syntax will alter the result of GetStdHandle.

Piping like that will cause Console.IsOutputRedirected to be true.

> Every time OutputHandle is called, there would be a PInvoke to GetStdHandle. Should the value be cahced?

We've opted not to cache it in the past so that it continues to respect any calls made to SetStdHandle. Caching it would be another breaking change.